### DISCUSSION TITLE
Help KRC 2 and Robots Plug in

### DISCUSSION THREAD

**Discussion by Ochinchilla** [visose/Robots#279] [Q&A] [answered]
Hello everyone,

I'm a student working with a 2005 Kuka K60 robot controlled by a KRC2 controller.

I've been using RoboDK for the past year, but now I want to integrate Grasshopper for more customization.

My issue arose when trying to post a .SRC file to the KRC2 controller. Previously, with RoboDK, I could copy a single .SRC file and execute it without problems. However, with the Robots plugin, it exports three files: one .DAT and two .SRC files. Is this normal? If so, which file should I run? Also, is there a way to export it as a single file?

I've read online that KRC2 controllers prefer shorter names. The "Robots" plugin exports files with complex names like "DWL8_T_ROB1" and "DWL8_T_ROB1_000" . Is there a way to simplify these names before exporting? For example, could they be saved as "DWL8A" and "DWL8B" instead?

Thank you for your assistance.

Best regards,
O

**Comment by visose** [comment]
Hi @Ochinchilla, yes the .DAT file is normal (having only an .SRC file without a .DAT file is the more unusual case).
You should copy all the generated files to the controller and run the main .SRC file.

The Robots plugin will warn you if you use a program name that is too long for a KRC2 controller. As long as you keep the length under the maximum allowed, making it even shorter has no other advantages.

**Reply by Ochinchilla** [reply]
Thank you for your reply.

To clarify, my robotic arm does not allow for copying an entire folder into the program folder. Should I copy all the files individually instead? If so, I will try this. Additionally, is there a way to specify in Robots that I am using a KRC2, or is this automatically assigned based on my robot configuration?

I also have another question. In RoboDK, I was able to add a program call instruction to start a spindle at 4000 RPM by typing SetRpm(4000). This would add that program call to my robotic program. Is there a similar feature available on Robots?

Thank you for your help. If the second question should be in a separate thread, please let me know.

Best regards,  
O

**Reply by visose** [reply, marked as answer]
You can select multiple files and copy them in one go.
The KRL code generated should be compatible with KRC2, no need to specify the controller version.
Use the "custom command" component to set the RPMs of your spindle.
